dask/_task_spec.py

```diff
@@ -611,18 +615,21 @@ def __init__(
         func: Callable,
         /,
         *args: Any,
+        _dependencies: set | frozenset | None = None,
```

Comment:
I suspect this is private because it shouldn't be used when instantiating new `Task` objects? This might be worth an inline comment.